common/persistence/sql/storage/sqldb/interfaces.go

```diff
@@ -589,51 +596,116 @@ type (
 		LockCurrentExecutions(filter *CurrentExecutionsFilter) (string, error)
 
 		InsertIntoTransferTasks(rows []TransferTasksRow) (sql.Result, error)
+		// SelectFromTransferTasks returns rows that match filter criteria from transfer_tasks table.
+		// Required filter params - {shardID, minTaskID, maxTaskID}
 		SelectFromTransferTasks(filter *TransferTasksFilter) ([]TransferTasksRow, error)
+		// DeleteFromTransferTasks deletes one or more rows from transfer_tasks table.
+		// Filter params - shardID is required. If TaskID is not nil, a single row is deleted.
+		// When MinTaskID and MaxTaskID are not-nil, a range of rows are deleted.
 		DeleteFromTransferTasks(filter *TransferTasksFilter) (sql.Result, error)
```

Comment:
An alternative would be to have two methods: DeleteTransferTask and DeleteTransferTaskRange with different filters.